misk/src/main/kotlin/misk/web/Request.kt

```diff
@@ -3,13 +3,53 @@ package misk.web
 import misk.web.actions.WebSocket
 import okhttp3.Headers
 import okhttp3.HttpUrl
+import okhttp3.MediaType
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.Response
+import okhttp3.ResponseBody
+import okio.BufferedSink
 import okio.BufferedSource
 import org.eclipse.jetty.http.HttpMethod
+import java.io.IOException
+import java.net.HttpURLConnection
 
 data class Request(
   val url: HttpUrl,
-  val method: HttpMethod,
+  val method: HttpMethod = HttpMethod.GET,
   val headers: Headers = Headers.of(),
   val body: BufferedSource,
   val websocket: WebSocket? = null
 )
+
+fun Request.toOkHttp3(newUrl: HttpUrl?): Response {
+  val client = OkHttpClient()
+
+  val okUrl = if (newUrl != null) newUrl else this.url
```

Comment:
is this ever non-null?

Reply:
Maybe we should have two methods, one to convert a Request to an okhttp3.Request, and a separate method to change that request’s URL? As-is this method does too many things

Reply:
Refactored into a few single purpose methods 👍

Reply:
yay